src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/RoleAssignmentFilter.java

```diff
@@ -309,18 +319,21 @@ private static Predicate searchByRegion(Root<TaskResource> root,
 
     private static Predicate searchByIncludingCaseId(Root<TaskResource> root,
                                                      CriteriaBuilder builder,
-                                                     RoleAssignment roleAssignment) {
-        String caseId = roleAssignment.getAttributes().get(CASE_ID.value());
-        if (StringUtils.hasText(caseId)) {
-            return builder.equal(root.get(CASE_ID_COLUMN), caseId);
+                                                     RoleAssignmentForSearch roleAssignment) {
+        Set<String> caseIds = roleAssignment.getCaseIds();
+        if (!CollectionUtils.isEmpty(caseIds)) {
+            if (caseIds.size() == ONE) {
```

Comment:
New line following re-engineering for performance. Can we add unit test to cover this line?